### PR TITLE
build fixes

### PR DESCRIPTION
**Pull request by ceolin**
Problems found when building using multilib

### CODE REVIEW

**Comment by glima** [comment]
+1

File: src/lib/common/Makefile [hunk 1]
```diff
@@ -53,8 +53,6 @@ obj-core-$(PLATFORM_CONTIKI) += \
 obj-core-$(PLATFORM_LINUX) += \
     sol-platform-linux-common.o \
     sol-board-detect.o
-obj-core-$(PLATFORM_LINUX)-extra-cflags += $(SYSTEMD_CFLAGS)
-obj-core-$(PLATFORM_LINUX)-extra-ldflags += $(SYSTEMD_LDFLAGS)
```
Comment (barbieri) [inline src/lib/common/Makefile]: just removing it doesn't look right. Maybe add a obj-core-$(PLATFORM_SYSTEMD)... ?

also, we need to link to systemd if either platform-systemd || sd-bus is used.

Comment (barbieri) [inline src/lib/common/Makefile]: ouch, just read more of that Makefile and right below it there is a block:
```
obj-core-$(PLATFORM_SYSTEMD)-extra-cflags += $(SYSTEMD_CFLAGS) $(UDEV_CFLAGS)
obj-core-$(PLATFORM_SYSTEMD)-extra-ldflags += $(SYSTEMD_LDFLAGS) $(UDEV_LDFLAGS)
```

But we may still fail if building linux-micro on a platform where sd-dbus is to be used.

Comment (ceolin) [inline src/lib/common/Makefile]: It is not a problem due:

obj-core-$(SOL_BUS)-extra-cflags += $(SYSTEMD_CFLAGS)
obj-core-$(SOL_BUS)-extra-ldflags += $(SYSTEMD_LDFLAGS)

Comment (dorileo) [inline src/lib/common/Makefile]: actually it's not a problem due to:

```make
obj-core-$(PLATFORM_SYSTEMD)-extra-cflags += $(SYSTEMD_CFLAGS) $(UDEV_CFLAGS)
obj-core-$(PLATFORM_SYSTEMD)-extra-ldflags += $(SYSTEMD_LDFLAGS) $(UDEV_LDFLAGS)
```

Comment (ceolin) [inline src/lib/common/Makefile]: the point is, if you disable systemd and enable sd-bus the target that will link with systemd is the SOL_BUS isn't ? Or are you saying that if disable systemd it still exists obj-core-$(PLATFORM_SYSTEMD) ?

Comment (dorileo) [inline src/lib/common/Makefile]: all right missed the sd-bus... @ceolin it's right..

Comment (barbieri) [inline src/lib/common/Makefile]: +1 then

**Comment by ceolin** [comment]
merged, thanks guys

**Pull request closed without merging**